internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -532,6 +540,29 @@ func fetchConfig(r runtime.Runtime) (out []byte, err error) {
 	return b, nil
 }
 
+func receiveConfigViaWebconfig(ctx context.Context, logger *log.Logger, r runtime.Runtime) ([]byte, error) {
+	cfgBytes, err := webconfig.Run(ctx, logger)
+	if err != nil {
+		return nil, fmt.Errorf("failed to receive config via webconfig: %w", err)
+	}
+
+	provider, err := configloader.NewFromBytes(cfgBytes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create config provider from webconfig bytes: %w", err)
+	}
+
+	if err = provider.Validate(r.State().Platform().Mode()); err != nil {
```

Comment:
just in terms of the UX, if we could validate in the `webconfig` and return error back via HTTP response so that user could retry, I think it might be easier

Reply:
I would strongly agree, but it appears that Validation is tied to the Platform.

Reply:
Okay, this is kind of weird.  There are two different Mode interfaces, but it looks like they are compatible.